### PR TITLE
Ref #574 adding warn message for queue length

### PR DESCRIPTION
**Pull request by Tanych**
Ref https://github.com/gwu-libraries/sfm-ui/issues/574
adding warning interval job to notification the sfm-admin check the harvester queue
Ref https://github.com/gwu-libraries/sfm-ui/issues/641
adding monitoring sfm-ui

### CODE REVIEW

File: sfm/sfm/settings/common.py [hunk 1]
```diff
@@ -209,6 +209,13 @@
 # sfm processing space threshold to send notification email,only ends with MB,GB,TB. eg. 500MB,10GB,1TB
 PROCESSING_THRESHOLD = env.get('PROCESSING_VOLUME_THRESHOLD', '10GB')
 
+# Whether to scan the amount of free space on /sfm-data and /sfm-processing
+PERFORM_MONITOR_QUEUE = env.get('SFM_PERFORM_MONITOR_QUEUE', 'True') == 'True'
+# frequency to check the queue message length,default is 2 hour
+MONITOR_QUEUE_HOUR = env.get('SFM_MONITOR_QUEUE_HOUR', '2')
+# queue threshold to send warning message
+QUEUE_LENGTH_THRESHOLD = env.get('QUEUE_LENGTH_THRESHOLD', '100')
```
Comment (justinlittman) [inline sfm/sfm/settings/common.py]: Set default to 10.

File: sfm/ui/notifications.py [hunk 2]
```diff
@@ -171,6 +172,48 @@ def _create_space_email(email_address, msg_cache):
...
+                log.error("Error sending email: %s", ex)
+
+
+def _create_queue_warn_email(email_address, msg_cache):
+    text_template = get_template('email/queue_length_email.txt')
+    html_template = get_template('email/queue_length_email.html')
+    msg_cache["url"] = _create_url(reverse('home'))
+    msg_cache["monitor_url"] = _create_url(reverse('monitor'))
+    d = Context(msg_cache)
+    msg = EmailMultiAlternatives("[WARNING] Long queue message on SFM server",
```
Comment (justinlittman) [inline sfm/ui/notifications.py]: "Long message queue" not "Long queue message"

File: sfm/ui/templates/email/queue_length_email.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{% load humanize %}
+<html>
+<body>
+<p>The following message queues on your <a href="{{ url }}">Social Feed Manager</a> server reach the warning level with <strong>{{ queue_threshold }}</strong> requests. </p>
```
Comment (justinlittman) [inline sfm/ui/templates/email/queue_length_email.html]: "have reached"

**Comment by justinlittman** [comment]
Upon further reflection, it will be necessary to be able to override on a per harvester/exporter basis.  This is particularly necessary for the web harvester, which should be expected to have a sizable queue length.

I'd recommend this be provided as a map in settings.  By default, the only value in that map can be for the web harvester, which should have it's own env setting.

**Comment by justinlittman** [comment]
One last request:  When there is a queue warning, can you put it on the home page for admins (similar to space usage).

**Comment by Tanych** [comment]
what I understand is that the warning message will only show when it has queue reach the warning level, otherwise it won;t show: it would look like:
new_admin_page

Should more simple?

**Comment by Tanych** [comment]
@justinlittman If it's too complex, I will simplify to show queue name.

**Comment by Tanych** [comment]
@justinlittman you can test the new view of the admin page.

**Pull request merged**